py/desispec/frame.py

```diff
@@ -89,7 +91,9 @@ def __init__(self, wave, flux, ivar, mask=None, resolution_data=None,
         if mask is None:
             self.mask = np.zeros(flux.shape, dtype=np.uint32)
         else:
-            self.mask = mask
+            if np.any(np.abs(np.asarray(mask)) > 2**32-1):
+                log.error('mask contains elements > 2**32-1; downcasting to 32-bit')
+            self.mask = np.asarray(mask, dtype=np.uint32)
```

Comment:
This looks like a different method to set the mask than in fiberflat.  Is there a reason you use `np.asarray()` here and not `mask.astype(np.uint32)`?